File: src/main/java/gameover/fwk/gdx/ai/Function2DPointComparator.java
package gameover.fwk.gdx.ai;

import com.badlogic.gdx.math.Vector2;

import java.util.Comparator;

/**
 * Created with IntelliJ IDEA.
 * User: Coin
 * Date: 23/07/14
 * Time: 16:31
 * To change this template use File | Settings | File Templates.
 */
public class Function2DPointComparator implements Comparator<Vector2> {
    private final float xSign;
    private final float ySign;

    public Function2DPointComparator(float xSign, float ySign) {
        this.xSign = xSign;
        this.ySign = ySign;
    }


    @Override
    public int compare(Vector2 o1, Vector2 o2) {
        float signum = Math.signum(o1.x - o2.x);
        if(signum==0){
            return (int) (Math.signum(o1.x - o2.x)*ySign);
        } else {
            return (int) (signum*xSign);
        }
    }
}

File: src/main/scala/gameover/fwk/libgdx/gfx/Function2DPointComparator.scala
package gameover.fwk.libgdx.gfx

import java.util.Comparator

import com.badlogic.gdx.math.Vector2

class Function2DPointComparator(xSign: Float, ySign: Float) extends Comparator[Vector2] {

  override def compare(o1: Vector2, o2: Vector2): Int = {
    val signum: Float = Math.signum(o1.x - o2.x)
    if (signum == 0) {
      (Math.signum(o1.x - o2.x) * ySign).toInt
    }
    else {
      (signum * xSign).toInt
    }
  }
}
